GoRogue/MapGeneration/PolygonArea.cs

```diff
@@ -155,14 +317,120 @@ private IEnumerable<IReadOnlyArea> GetBoundariesContaining(int x, int y)
 
         /// <inheritdoc/>
         public bool Intersects(IReadOnlyArea area) => _points.Intersects(area);
+        #endregion
 
-        /// <inheritdoc/>
-        public Rectangle Bounds => _points.Bounds;
+        #region Transformation
 
-        /// <inheritdoc/>
-        public int Count => _points.Count;
+        /// <summary>
+        /// Moves the Polygon in the indicated direction.
+        /// </summary>
+        /// <param name="dx">The X-value by which to shift</param>
+        /// <param name="dy">The Y-value by which to shift</param>
+        /// <returns></returns>
+        public PolygonArea Translate(int dx, int dy)
+            => Translate(new Point(dx, dy));
 
-        /// <inheritdoc/>
-        public Point this[int index] => _points[index];
+        /// <summary>
+        /// Moves the Polygon in the indicated direction.
+        /// </summary>
+        /// <param name="delta">The amount (X and Y) to translate by.</param>
+        /// <returns>A new, translated PolygonArea</returns>
+        public PolygonArea Translate(Point delta)
+        {
+            var corners = new Point[Corners.Count];
```

Comment:
Because we provide a constructor that takes a `List` by reference and avoids copying the list, it would probably be more efficient here to allocate this as a `List`, then we can pass it with the `ref` keyword to the `PolygonArea` constructor.

This does beg one other question, which is whether the `_corners` property of `PolygonArea` should in fact be a `List`, or an array?  It likely should only be a list if there is a need to use it as such; an array will cut _some_ overhead and also express intent more clearly.

If it became an array, the parameter that takes `ref List<Point>` would of course have to be changed to `ref Point[]`, which has implications for some of the other comments.